examples/squiral/main.go

```diff
@@ -0,0 +1,341 @@
+		s.rot = (s.rot + 1) % 2
+	}
+
+	// 1. try to advance the spiral in its rotation
+	// direction (clockwise or counter-clockwise).
+	for _, next := range rotors[s.rot] {
+		dir := (s.dir + next) % 4
+		off := dirs[dir]
+		// Peek all targets by priority.
+		target := vec2{
+			x: x + off.x,
+			y: y + off.y,
+		}
+		if auto.colorMap[target.x][target.y] == free {
+			// If the target is free we need to also check the
+			// surrounding cells.
+
+			// a. Test if next cell in direction dir does not have
+			// the same color as this squiral.
+			ntarg := vec2{
+				x: target.x + off.x,
+				y: target.y + off.y,
+			}
+			if auto.colorMap[ntarg.x][ntarg.y] == s.col {
+				// If this has the same color, we cannot go into this direction,
+				// to avoid ugly blocks of equal color.
+				continue // try next direction
+			}
+
+			// b. Test all outer fields for the color of the
+			// squiral itself.
+			horivert := dir % 2
+			xtarg := vec2{}
+			set := true
+			for _, out := range outers[horivert] {
+				xtarg.x = target.x + out.x
+				xtarg.y = target.y + out.y
+
+				// If one of the outer targets equals the squiral's
+				// color, again continue with next direction.
+				if auto.colorMap[xtarg.x][xtarg.y] == s.col {
+					// If this is not free we cannot go into this direction.
+					set = false
+					break // try next direction
+				}
+
+				xtarg.x = ntarg.x + out.x
+				xtarg.y = ntarg.y + out.y
+
+				// If one of the outer targets equals the squiral's
+				// color, again continue with next direction.
+				if auto.colorMap[xtarg.x][xtarg.y] == s.col {
+					// If this is not free we cannot go into this direction.
+					set = false
+					break // try next direction
+				}
+			}
+
+			if set {
+				s.pos = target
+				s.dir = dir
+				// 2. set the color of this squiral to its
+				// current position.
+				setpix(s.pos, s.col)
+				return
+			}
+		}
+	}
+
+	s.dead = true
+}
+
+type automaton struct {
+	squirals [numOfSquirals]squiral
+	colorMap [width][height]color.Color
+}
+
+func (au *automaton) init() {
+	// Init the test grid with color (0,0,0,0) and the borders of
+	// it with color(0,0,0,254) as a blocker color, so the squirals
+	// cannot escape the scene.
+	for x := 0; x < width; x++ {
+		for y := 0; y < height; y++ {
+			if x == 0 || x == width-1 || y == 0 || y == height-1 {
+				auto.colorMap[x][y] = blocker
+			} else {
+				auto.colorMap[x][y] = free
+			}
+		}
+	}
+
+	for i := 0; i < numOfSquirals; i++ {
+		auto.squirals[i].spawn()
+	}
+}
+
+func (au *automaton) step() {
+	for i := 0; i < numOfSquirals; i++ {
+		for s := 0; s < au.squirals[i].speed; s++ {
+			au.squirals[i].step(i)
+			if au.squirals[i].dead {
+				au.squirals[i].spawn()
+			}
+		}
+	}
+}
+
+func setpix(xy vec2, col color.Color) {
+	canvas.Set(xy.x, xy.y, col)
+	auto.colorMap[xy.x][xy.y] = col
+}
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+	c, err := ebiten.NewImage(width, height, ebiten.FilterNearest)
```

Comment:
You can ignore the error form `NewImage`. This is always nil. This error type is just for backward compatibility :-)

Also please use `ebiten.FilterDefault`.

Then

```
canvas, _ = ebiten.NewImage(width, height, ebiten.FilterDefault)
```

is fine.